server/src/service.rs

```diff
@@ -114,16 +114,70 @@ impl Service {
     }
 
     #[zbus(out_args("collection"))]
-    pub async fn read_alias(&self, _name: &str) -> Result<ObjectPath, ServiceError> {
-        todo!()
+    pub async fn read_alias(
+        &self,
+        name: &str,
+        #[zbus(object_server)] object_server: &zbus::ObjectServer,
+    ) -> Result<OwnedObjectPath, ServiceError> {
+        let collections = self.collections.lock().await;
+
+        for collection in collections.iter() {
+            let collection_ifce_ref = object_server.interface::<_, Collection>(collection).await?;
+            let collection = collection_ifce_ref.get_mut().await;
+
+            if name == "login" {
+                if collection.label().await == name {
+                    drop(collections);
```

Comment:
No need to drop the mutex manually, it will be dropped at the end of the scope, this is not C.

Reply:
Removed.